### ISSUE TITLE
Build fails for current Stackage nightly

### ISSUE THREAD

**Issue by ocramz** [Soostone/retry#51]
We might want to relax the upper dependency bounds, if possible:

```
2.26s$ stack $ARGS test --no-terminal --haddock --no-haddock-deps
Selected resolver: nightly-2017-08-30

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for retry-0.7.4.2:

 HUnit-1.6.0.0 must match >=1.2.5.2 && <1.6 (latest applicable is 1.5.0.0)
 QuickCheck-2.10.0.1 must match >=2.7 && <2.10 (latest applicable is 2.9.2)

needed since retry-0.7.4.2 is a build target.
```

**Comment by MichaelXavier** [comment]
Resolved by #52. Thanks!

**Comment by MichaelXavier** [comment]
Released at 0.7.4.3.
